config/routes.rb

```diff
@@ -103,4 +103,17 @@
       post 'openscap_proxy_changed'
     end
   end
+
+  namespace :api do
+    scope "(:api_version)", :module => :v2,
+                            :defaults => { :api_version => 'v2' },
+                            :api_version => /v2/,
+                            :constraints => ApiConstraints.new(:version => 2, :default => true) do
+      resources :hosts, :only => [] do
```

Comment:
this needs to be wrapped in `constraints(:id => /[^\/]+/) do ... end `

Reply:
Updated with constraints.